File: tests/fixtures/lang_samples/kotlin/assertions.kt
// Fixture for assertion-block detection (test-quality smells).

fun testManyAsserts() {
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
    assertEquals(1, 1)
}

fun testFewAsserts() {
    assertEquals(1, 1)
    val x = 2
    assertEquals(x, 2)
}

File: tests/fixtures/lang_samples/cpp/assertions.cpp
// Fixture for assertion-block detection (test-quality smells).
// GoogleTest-style EXPECT_*/ASSERT_* macros are ordinary call expressions.

void testManyAsserts() {
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
    EXPECT_EQ(1, 1);
}

void testFewAsserts() {
    EXPECT_EQ(1, 1);
    int x = 2;
    EXPECT_EQ(x, 2);
}
